controllers/hetznercluster_controller_test.go

```diff
@@ -848,90 +848,102 @@ func TestSetControlPlaneEndpoint(t *testing.T) {
 		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: nil}}
 		got := SetControlPlaneEndpoint(hetznerCluster)
 		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == true){
-			t.Fatal("Wrong value for ControlPlaneLoadBalancer.Enabled")
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
 		}
 		if (hetznerCluster.Spec.ControlPlaneEndpoint != nil){
-			t.Fatal("Wrong value for hetznerCluster.Spec.ControlPlaneEndpoint")
+			t.Fatalf("Wrong value for hetznerCluster.Spec.ControlPlaneEndpoint. Got: %s, Want: '<nil>'", hetznerCluster.Spec.ControlPlaneEndpoint)
 		}
 		assert.Equal(t, false, got)
 	})
 	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is false but hetznerCluster.Spec.ControlPlaneEndpoint is not nil", func(t *testing.T) {
 		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: &clusterv1.APIEndpoint{Host: "xyz"}}}
 		got := SetControlPlaneEndpoint(hetznerCluster)
-		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == true){
-			t.Fatal("Wrong value for ControlPlaneLoadBalancer.Enabled")
+		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled != false){
+			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
 		}
 		if (hetznerCluster.Spec.ControlPlaneEndpoint == nil){
-			t.Fatal("ControlPlaneEndpoint should not be nil")
+			t.Fatalf("ControlPlaneEndpoint should not be nil. Got: %s, Want: (Host: 'xyz')", hetznerCluster.Spec.ControlPlaneEndpoint)
```

Comment:
here you can make it easier: "ControlPlaneEndpoint must not be nil" without anything else, as you had before